include/extractor/io.hpp

```diff
@@ -96,6 +99,51 @@ inline void write(const boost::filesystem::path &path, const SegmentDataContaine
     writer.WriteFrom(segment_data.rev_durations);
     writer.WriteFrom(segment_data.datasources);
 }
+
+// read/write for conditional turn restrictions file
+inline void read(const boost::filesystem::path &path, std::vector<InputRestrictionContainer> &restrictions)
+{
+    const auto fingerprint = storage::io::FileReader::VerifyFingerprint;
+    storage::io::FileReader reader{path, fingerprint};
+
+    auto num_indices = reader.ReadElementCount32();
+    restrictions.resize(num_indices);
+    while (num_indices > 0)
+    {
+        InputRestrictionContainer res;
+        bool is_only;
+        reader.ReadInto(res.restriction.via);
+        reader.ReadInto(res.restriction.from);
+        reader.ReadInto(res.restriction.to);
+        reader.ReadInto(is_only);
+        for (auto &c : res.restriction.condition)
+        {
+            reader.ReadInto(c.modifier);
+            reader.DeserializeVector(c.times);
+            reader.DeserializeVector(c.weekdays);
+            reader.DeserializeVector(c.monthdays);
+        }
+        res.restriction.flags.is_only = is_only;
+        restrictions.push_back(res);
```

Comment:
You can `std::move(res)` here to move the restriction container into the vector. At the moment we make a copy of res and then destroy res immediately after. Similar, you could put the `InputRestrictionContainer` above outside of the loop, otherwise we create an instance of it for every iteration.